src/lib.rs

```diff
@@ -234,25 +234,27 @@ pub trait PolynomialCommitment<F: Field>: Sized {
     /// TODO: as a high-level function, we will remove the fs_rng.
     fn check<'a>(
         vk: &Self::VerifierKey,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
+        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>> + Clone,
         point: F,
         values: impl IntoIterator<Item = F>,
         proof: &Self::Proof,
         rng: Option<&mut dyn RngCore>,
-        fs_rng: &mut Self::RandomOracle,
     ) -> Result<bool, Self::Error>
         where
             Self::Commitment: 'a,
     {
-        // as in open(), setup Fiat-Shamir rng, etc.
+        let mut fs_rng = Self::RandomOracle::from_seed(&to_bytes![
+            point,
+            commitments.clone().into_iter().collect::<Vec<_>>()
+        ].unwrap());
```

Comment:
as for fn open()

Reply:
Previous functions state is restored as Daniele requested bellow